xds/src/main/java/io/grpc/xds/client/ControlPlaneClient.java

```diff
@@ -238,7 +238,7 @@ boolean isInError() {
    */
   // Must be synchronized.
   void readyHandler(boolean shouldSendInitialRequest) {
-    if (shouldSendInitialRequest) {
+    if (!shutdown && shouldSendInitialRequest) {
```

Comment:
This is already checked in `onReady()`. This is why I was wanting the callback methods not to be split, because it isn't otherwise clear which checks each method should do. Note that in general `!shutdown` is a really weak check, because it doesn't check that this is the current ADS stream. So I see it as a code smell.

Reply:
Removed